raiden_contracts/deploy/etherscan_verify.py

```diff
@@ -41,108 +41,94 @@
-    elif chain_id == 4:
-        etherscan_api = 'https://api-rinkeby.etherscan.io/api'
-    elif chain_id == 42:
-        etherscan_api = 'https://api-kovan.etherscan.io/api'
-    elif chain_id == 1:
-        etherscan_api = 'https://api.etherscan.io/api'
-    else:
-        click.echo(f"Unknown chain_id {chain_id}", err=True)
-        exit(1)
-
-    deployment_info = get_contracts_deployed(chain_id)
-
+def etherscan_verify(
+        chain_id: int,
+        apikey: str,
+        guid: Optional[str],
+        contract_name: Optional[str],
+):
     if guid:
-        guid_status(etherscan_api, guid)
+        guid_status(api_of_chain_id(chain_id), guid)
         return
 
     if contract_name is None or contract_name == CONTRACT_ENDPOINT_REGISTRY:
-        path = source_path['raiden'].joinpath(f'{CONTRACT_ENDPOINT_REGISTRY}.sol')
-        source = path.read_text()
-        constructor_args = ''
-        etherscan_verify_contract(
-            etherscan_api,
-            apikey,
-            contract_manager.contracts[CONTRACT_ENDPOINT_REGISTRY],
-            CONTRACT_ENDPOINT_REGISTRY,
-            deployment_info['contracts'][CONTRACT_ENDPOINT_REGISTRY],
-            source,
-            constructor_args,
-        )
+        etherscan_verify_contract(chain_id, apikey, 'raiden', CONTRACT_ENDPOINT_REGISTRY)
 
     if contract_name is None or contract_name == CONTRACT_SECRET_REGISTRY:
-        path = source_path['raiden'].joinpath(f'{CONTRACT_SECRET_REGISTRY}.sol')
-        source = path.read_text()
-        constructor_args = ''
-        etherscan_verify_contract(
-            etherscan_api,
-            apikey,
-            contract_manager.contracts[CONTRACT_SECRET_REGISTRY],
-            CONTRACT_SECRET_REGISTRY,
-            deployment_info['contracts'][CONTRACT_SECRET_REGISTRY],
-            source,
-            constructor_args,
-        )
+        etherscan_verify_contract(chain_id, apikey, 'raiden', CONTRACT_SECRET_REGISTRY)
 
     if contract_name is None or contract_name == CONTRACT_TOKEN_NETWORK_REGISTRY:
-        joined_file = Path(__file__).parent.joinpath('joined.sol')
-        command = [
-            './utils/join-contracts.py',
-            '--import-map',
-            '{"raiden": "contracts", "test": "contracts/test", "services": "contracts/services"}',
-            'contracts/TokenNetworkRegistry.sol',
-            str(joined_file),
-        ]
-        old_working_dir = Path.cwd()
-        chdir(Path(__file__).parent.parent)
-        try:
-            combiner = subprocess.Popen(command)
-            combiner.wait()
-        finally:
-            chdir(old_working_dir)
-
-        source = joined_file.read_text()
-
-        constructor_arguments = deployment_info['contracts'][
-            CONTRACT_TOKEN_NETWORK_REGISTRY
-        ]['constructor_arguments']
-        abi = contract_manager.contracts[CONTRACT_TOKEN_NETWORK_REGISTRY]['abi']
+        etherscan_verify_contract(chain_id, apikey, 'raiden', CONTRACT_TOKEN_NETWORK_REGISTRY)
```

Comment:
Those lines still look very similar. May I suggest
```python
if contract_name is None:
    contracts = [CONTRACT_ENDPOINT_REGISTRY, CONTRACT_SECRET_REGISTRY, CONTRACT_TOKEN_NETWORK_REGISTRY]
else:
    contracts = [contract_name]
for contract in contracts:
    etherscan_verify_contract(chain_id, apikey, 'raiden', contract)
```

Maybe you could even use
```python
@click.option(
    '--contract-name',
    multiple=True,
    default=[CONTRACT_ENDPOINT_REGISTRY, CONTRACT_SECRET_REGISTRY, CONTRACT_TOKEN_NETWORK_REGISTRY]
)
...
for contract in contract_name:
    etherscan_verify_contract(chain_id, apikey, 'raiden', contract)
```

Reply:
The one with `@click.option` is very appealing.  I created an issue for that https://github.com/raiden-network/raiden-contracts/issues/535